openedx/core/djangoapps/content_tagging/rest_api/v1/filters.py

```diff
@@ -2,20 +2,95 @@
 API Filters for content tagging org
 """
 
+from django.db.models import Exists, OuterRef, Q
 from rest_framework.filters import BaseFilterBackend
 
 import openedx_tagging.core.tagging.rules as oel_tagging
+from organizations.models import Organization
+
+from ...rules import get_admin_orgs, get_content_creator_orgs, get_instructor_orgs
+from ...models import TaxonomyOrg
 
 
 class UserOrgFilterBackend(BaseFilterBackend):
     """
+    Filter taxonomies based on user's orgs roles
+
     Taxonomy admin can see all taxonomies
-    Everyone else can see only enabled taxonomies
+    Org staff can see all taxonomies from their orgs
+    Content creators and instructors can see enabled taxonomies avaliable to their orgs
+    """
+
+    def filter_queryset(self, request, queryset, _):
+        if oel_tagging.is_taxonomy_admin(request.user):
+            return queryset
 
+        orgs = list(Organization.objects.all())
+        user_admin_orgs = get_admin_orgs(request.user, orgs)
+        content_creator_orgs = get_content_creator_orgs(request.user, orgs)
+        instructor_orgs = get_instructor_orgs(request.user, orgs)
+        user_orgs = list(set(content_creator_orgs) | set(instructor_orgs))
```

Comment:
Suggestion: Since we're always combining these two sets of orgs, could they be combined into a single method? That reduces somewhat the amount of rules knowledge present in these filters, e.g.

```suggestion
        user_orgs = get_user_orgs(request.user, orgs) # returns set of content_creator + instructor orgs
```

Reply:
Good catch!
Done: c29f48f8a871efd532d6bbf85ba06dcd18ba8cc6